Some army scripts

Mercule opened the pull request.

Mercule committed:
Some army scripts

Mercule committed:
Merge branch 'Skua' of https://github.com/Mercule/Scripts into Skua

Mercule committed:
Some army scripts

Mercule committed:
Some army scripts

Mercule committed:
Add files via upload

Mercule committed:
Update ArmyTaintedGem.cs

LordExelot commented on Army/ArmyBloodMoonToken.cs:
```diff
@@ -0,0 +1,64 @@
+//cs_include Scripts/CoreBots.cs
+//cs_include Scripts/CoreFarms.cs
+//cs_include Scripts/CoreAdvanced.cs
```
These go unused

LordExelot commented on Army/ArmyBloodMoonToken.cs:
```diff
@@ -0,0 +1,64 @@
+{
+    private IScriptInterface Bot => IScriptInterface.Instance;
+    private CoreBots Core => CoreBots.Instance;
+    private CoreFarms Farm = new();
+    private CoreAdvanced Adv => new();
```
These go unused

LordExelot commented on Army/ArmyBloodMoonToken.cs:
```diff
@@ -0,0 +1,64 @@
+        Core.BankingBlackList.AddRange(Loot);
+
+        Core.SetOptions(disableClassSwap: true);
+        bot.Options.RestPackets = false;
```
Beauty thing, but Bot with capital b

Mercule replied:
For this, I just find out if all army scripts are using b instead B xD

LordExelot commented on Army/ArmyDarkonErrands.cs:
```diff
@@ -0,0 +1,99 @@
+    public void Setup(Method Method, int quant = 222)
+    {
+        Core.EquipClass(ClassType.Solo);
+        if (Method.ToString() == "Third_Errands")
```
I recommend else if for these stacks

LordExelot commented on Army/ArmyMirrorRealmToken.cs:
```diff
@@ -0,0 +1,72 @@
+        }
+        if (Method.ToString() == "Kill_Boss")
+        {
+        Core.EquipClass(ClassType.Solo);
+        Core.RegisterQuests(3188);
+        Army.SmartAggroMonStart("mirrorportal", "Chaos Harpy");
```
Missing indent

LordExelot commented on Army/Rep/ArmyGoodEvilRep.cs:
```diff
@@ -0,0 +1,81 @@
+
+    public void ScriptMain(IScriptInterface bot)
+    {
+
```
White line

LordExelot reviewed (commented):
Left a bit of feedback, most of the feedback I left you is applicable to almost every other bot, so make sure that when going through the feedback, you also look if the other bots have the same issue

Mercule committed:
Update

Mercule committed:
Merge branch 'BrenoHenrike:Skua' into Skua

Mercule committed:
Merge branch 'Skua' of https://github.com/Mercule/Scripts into Skua

LordExelot commented on Army/ArmyDarkonErrands.cs:
```diff
@@ -0,0 +1,101 @@
+//cs_include Scripts/CoreBots.cs
+//cs_include Scripts/CoreFarms.cs
+//cs_include Scripts/CoreAdvanced.cs
```
Unused

LordExelot commented on Army/ArmyDarkonErrands.cs:
```diff
@@ -0,0 +1,101 @@
+//cs_include Scripts/CoreFarms.cs
+//cs_include Scripts/CoreAdvanced.cs
+//cs_include Scripts/Army/CoreArmyLite.cs
+//cs_include Scripts/CoreStory.cs
```
Unused

LordExelot commented on Army/ArmyDarkonErrands.cs:
```diff
@@ -0,0 +1,101 @@
+        ItemBase[] QuestReward = QuestData.Rewards.ToArray();
+
+        Core.AddDrop(item);
+        if (!Bot.Quests.Active.Contains(QuestData))
```
Check is redundant

LordExelot commented on Army/ArmyMirrorRealmToken.cs:
```diff
@@ -0,0 +1,74 @@
+//cs_include Scripts/CoreBots.cs
+//cs_include Scripts/CoreFarms.cs
+//cs_include Scripts/CoreAdvanced.cs
```
Unused

LordExelot commented on Army/ArmyMirrorRealmToken.cs:
```diff
@@ -0,0 +1,74 @@
+            Army.SmartAggroMonStart("overworld", "Undead Bruiser", "Undead Mage", "Undead Minion");
+        }
+
+        if (Method.ToString() == "Kill_Boss")
```
Else if

LordExelot commented on Army/ArmyMirrorRealmToken.cs:
```diff
@@ -0,0 +1,74 @@
+        Army.AggroMonStop(true);
+    }
+
+    private string[] Loot = { "Mirror Realm Token" };
```
Kind of a waste of memory to save a single item in a string array, just have the token in the banking black list and adddrop instead

LordExelot commented on Army/Rep/ArmyGoodEvilRep.cs:
```diff
@@ -0,0 +1,80 @@
+//cs_include Scripts/CoreBots.cs
+//cs_include Scripts/CoreFarms.cs
+//cs_include Scripts/CoreAdvanced.cs
```
Unused

LordExelot commented on Army/Rep/ArmyGoodEvilRep.cs:
```diff
@@ -0,0 +1,80 @@
+
+    public void rank4()
+    {
+    Core.RegisterQuests(364, 369);
+    Army.SmartAggroMonStart("swordhavenbridge", "Slimes");
+            Bot.Combat.Attack("*");
+    
+    Army.AggroMonClear();
+    Core.CancelRegisteredQuests();
```
Indent is fucked

LordExelot commented on Army/Rep/ArmyGoodEvilRep.cs:
```diff
@@ -0,0 +1,80 @@
+
+    public void rankMAX()
+    {
+    Core.RegisterQuests(367, 372);
+    Army.SmartAggroMonStart("castleundead", "Skeletal Viking", "Skeletal Warrior");
+            Bot.Combat.Attack("*");
+
+    Army.AggroMonStop(true);
+    Core.CancelRegisteredQuests();
```
Indent is fucked

LordExelot reviewed (commented):
Hey, i left a few new comments, but mostly restated previously made comments that weren't handled yet

Mercule committed:
Merge branch 'BrenoHenrike:Skua' into Skua

Mercule committed:
Another update scripts

For darkon it have 4 options, so you can choose option depend on your comp

For good evil, it farm both rep at same time

Mercule committed:
Merge branch 'Skua' of https://github.com/Mercule/Scripts into Skua

LordExelot commented on Army/Rep/ArmyGoodEvilRep.cs:
```diff
@@ -0,0 +1,77 @@
+        Core.EquipClass(ClassType.Farm);
+
+        while (!Bot.ShouldExit && Farm.FactionRank("Evil") < 4)
+        rank4();
```
+indent

LordExelot reviewed (approved):
If you could fix the indent on the rank4's and rankMax's, were good to go

Mercule committed:
Merge branch 'BrenoHenrike:Skua' into Skua

Mercule committed:
Update ArmyGoodEvilRep.cs

add indent

Mercule committed:
Merge branch 'Skua' of https://github.com/Mercule/Scripts into Skua
